### PR TITLE
Improve channel tracking for gRPC authentication traces

### PR DESCRIPTION
**Pull request by ggezer**
Main purpose of this PR is to make gRPC authentication tracing to contain channel information in order to improve debugging.

### CODE REVIEW

File: core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java [hunk 1]
```diff
@@ -88,43 +93,44 @@ public void setServerObserver(StreamObserver<SaslMessage> requestObserver) {
...
       mHandshakeFuture.setException(e);
-      mRequestObserver
-          .onError(Status.fromCode(Status.Code.UNAUTHENTICATED).withCause(e).asException());
+      mRequestObserver.onError(e);
     }
   }
 
   @Override
   public void onError(Throwable throwable) {
+    LOG.warn("Received error on client driver for channel: {}. Error: {}", mChannelId, throwable);
```
Comment (ggezer) [inline core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java]: Reverting to WARN, here and at `SaslStreamServerDriver` as it proved very useful in a recent error investigation.

**Comment by ggezer** [comment]
@madanadit PTAL. Thanks!

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Comment by madanadit** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**